trace/src/main/org/jetbrains/lincheck/trace/TraceRecorderTracePoints.kt

```diff
@@ -27,18 +27,54 @@ import java.io.DataInput
 import java.io.DataOutput
 import java.math.BigDecimal
 import java.math.BigInteger
+import java.util.EnumSet
 import java.util.concurrent.atomic.AtomicInteger
+import kotlin.experimental.or
 import kotlin.reflect.KClass
 
 private val EVENT_ID_GENERATOR = AtomicInteger(0)
 
 var INJECTIONS_VOID_OBJECT: Any? = null
 
+/**
+ * Describes status of tracepoint in trace diff
+ */
+enum class DiffStatus {
+    /**
+     * Trace is not diff at all, or this tracepoint is identical in two traces.
+     */
+    COMMON,
+
+    /**
+     * This tracepoint was found in the first, but not in the second trace, when diff was created.
+     */
+    REMOVED,
+
+    /**
+     * This tracepoint was not found in the first, but found in the second trace, when diff was created.
+     */
+    ADDED,
+
+    /**
+     * This is call tracepoint, and it has the same method in both traces but differs in arguments values.
+     */
+    ARGS_DIFFERENCE,
```

Comment:
Let's make it more future proof. We can name it `EDITED` where edited has open meaning: it could be arguments change, or could be some other change of trace point payload (without structural change).